src/common/utils/resolve_reference.py

```diff
@@ -10,6 +10,31 @@
 from storage.data_source_class import DataSource
 
 
+def split_reference(reference: str) -> list[str]:
+    """Splits a reference into it's string components
+    e.g. "$123.content[1]" -> ["$123", ".content", "[1]"]"""
+    parts = []
+    remaining_ref = reference
+    while remaining_ref:
+        if remaining_ref[0] in "./":
+            prefix_delim = remaining_ref[0]
+            content = re.split(r"[./\[(]", remaining_ref, 2)[1]
```

Comment:
What if `reference=="/[(_id=1)].content[(name=myCarRental)].cars[0]"`. Then `content==""`. Perhaps add some tests for this?

Reply:
fixed in https://github.com/equinor/data-modelling-storage-service/pull/435